src/main/java/com/autotune/analyzer/recommendations/engine/RecommendationEngine.java

```diff
@@ -1984,213 +1979,255 @@ public void fetchMetricsBasedOnProfileAndDatasource(KruizeObject kruizeObject, T
             }
 
             String maxDateQuery = null;
-            List<Metric>metrics = metricProfile.getSloInfo().getFunctionVariables();
-            for (Metric metric: metrics) {
-                String name = metric.getName();
-                if(name.equals("maxDate")){
-                    String query = metric.getAggregationFunctionsMap().get("max").getQuery();
-                    maxDateQuery = query;
-                    break;
-                }
+            String experimentType = kruizeObject.getExperimentType();
+            if (null == experimentType || AnalyzerConstants.ExperimentTypes.CONTAINER_EXPERIMENT.equalsIgnoreCase(experimentType)) {
```

Comment:
avoid having this complicated check mutiple times 
instead have function inside KruizeObject and use it every where

```
if (ko.isContainerExperiment()) {

else if (ko.isNamespaceExperiment()) {
```

Reply:
Updated as discussed.